core/src/main/java/org/elasticsearch/ingest/processor/SplitProcessor.java

```diff
@@ -60,7 +63,9 @@ public void execute(IngestDocument document) {
         if (oldVal == null) {
             throw new IllegalArgumentException("field [" + field + "] is null, cannot split.");
         }
-        document.setFieldValue(field, Arrays.asList(oldVal.split(separator)));
+        List<String> splitList = new ArrayList<>();
```

Comment:
you mean providing the size of the array I guess? cause I don't see a constructor that accepts an array in ArrayList.